### PR TITLE
pyocf tests for start/stop

### PR DESCRIPTION
**Pull request by Deixx**
Tests for starting cache and checking if cache mode behaves properly
Tests for stopping cache with and without flush

Signed-off-by: Daniel Madej <daniel.madej@intel.com>

---
This change is [Reviewable](https://reviewable.io/reviews/open-cas/ocf/113)

### CODE REVIEW

**Comment by KamilLepek** [comment]
retest this please

File: tests/functional/pyocf/types/cache.py [hunk 1]
```diff
@@ -383,6 +383,7 @@ def get_stats(self):
                 "core_count": cache_info.core_count,
                 "metadata_footprint": Size(cache_info.metadata_footprint),
                 "metadata_end_offset": Size(cache_info.metadata_end_offset),
+                "cache_id": self.owner.lib.ocf_cache_get_id(self),
```
Comment (imjfckm) [inline tests/functional/pyocf/types/cache.py]: At this point cache is unlocked - there's a possibility that it's already gone.

File: tests/functional/tests/management/test_start_stop.py [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+    io.configure(offset, data.size, IoDir.WRITE, 0, 0)
+    io.set_queue(exported_obj.cache.get_default_queue())
+
+    completion = OcfCompletion([("err", c_int)])
+    io.callback = completion.callback
+    io.submit()
+    completion.wait()
+
+    logger.info("[ASSERT] IO to exported object completion")
+    assert completion.results["err"] == 0
```
Comment (imjfckm) [inline tests/functional/tests/management/test_start_stop.py]: Assert supports syntax which includes explanation for it, this way you don't have to write to logs and have it show up nicely in pytest fail logs:
`assert <condition>, "Message for the assert"`
Picked a random assert for this comment, but it applies to (nearly?) all asserts in this file

File: tests/functional/pyocf/types/cache.py (hunk 1 above)
Comment (Deixx) [reply tests/functional/pyocf/types/cache.py]: Done

File: tests/functional/tests/management/test_start_stop.py (hunk 2 above)
Comment (Deixx) [reply tests/functional/tests/management/test_start_stop.py]: Done

**Comment by KamilLepek** [comment]
retest this please

**Comment by KamilLepek** [comment]
retest this please

**Comment by KamilLepek** [comment]
retest this please

**Comment by KamilLepek** [comment]
retest this please

**Pull request merged**